crates/nodeapi/src/lib.rs

```diff
@@ -167,18 +181,32 @@ pub async fn health_check(api: web::types::State<NodeApi>) -> impl Responder {
 }
 
 #[allow(unused)]
-pub async fn list_services(api: web::types::State<NodeApi>) -> impl Responder {
-    let services = &api.services;
+pub async fn list_services(api: web::types::State<Arc<Mutex<NodeApi>>>) -> impl Responder {
```

Comment:
This should be private. Also, let's remove the `#[allow(unused)]`

Reply:
done